Popup docs (#1877)
telerik/blazor-docs: components/popup/overview.md

---
title: Overview
page_title: Popup Overview
description: Discover the Blazor Popup. Learn how to add the component to your app and explore its features like positioning, alignment, and animation customization.
slug: popup-overview
tags: telerik,blazor,popup
published: True
position: 0
---

# Blazor Popup Overview

> The <a href = "https://www.telerik.com/blazor-ui/popup" target="_blank">Blazor Popup component</a> helps you easily display a popup for a target element (anchor) in your application. You can use to Telerik UI for Blazor Popup to display additional information. This article explains how to start using the component and describes its features.

Reviewer: ```suggestion
The <a href = "https://www.telerik.com/blazor-ui/popup" target="_blank">Blazor Popup component</a> helps you easily display a popup for a target element (anchor) in your application. You can use the Telerik UI for Blazor Popup to display additional information. This article explains how to start using the component and describes its features.
```

| `VerticalAlign` | `PopupVerticalAlign ` enum <br /> (`Top`) | Determines if the Popup will touch the anchor with its top, bottom, or center. [Read more about Popup Positioning.]({%slug popup-position-collision%}) |
| `VerticalCollision` | `PopupCollision  ` enum <br /> (`Flip`) | Sets the behavior of the Popup when it doesn't fit in the viewport based on the vertical plane. [Read more about Popup collision behavior.]({%slug popup-position-collision%}) |
| `VerticalOffset` | `double` | Sets the vertical space between the Popup and its anchor in `pixels`. |

### Styling and Appearance

The following parameters enable you to customize the appearance of the Blazor Popup:

| Parameter | Type | Description |
| --- | --- | --- |
| `Class` | `string` | The custom CSS class to be rendered on the `<div>` element, which wraps the component `ChildContent`. Use for [styling customizations]({%slug themes-override%}). |
| `Height` | `string` | The height of the Popup. |
| `Width` | `string` | The width of the Popup. |

## Popup Reference and Methods

To execute Popup methods, obtain a reference to the component instance with `@ref`.

| Method  | Description |
|---------|-------------|

> | `Refresh` | Use this method to programmatically re-render the Popup. <br /> The Popup is rendered as a child of the `TelerikRootComponent`, instead of where it is declared. As a result, it doesn't automatically refresh when its content is updated. In such cases, the `Refresh` method comes in handy to ensure that the Popup content is up-to-date. |

Reviewer: Note that in this case you don't need the extra spaces around the `<br />` element.

---
title: Overview
page_title: Popup Overview
description: Discover the Blazor Popup. Learn how to add the component to your app and explore its features like positioning, alignment, and animation customization.
slug: popup-overview
tags: telerik,blazor,popup
published: True
position: 0
---

# Blazor Popup Overview

> The <a href = "https://www.telerik.com/blazor-ui/popup" target="_blank">Blazor Popup component</a> helps you to easily add popups to an element in your application. You can use to Telerik UI for Blazor Popup to display additional information. This article explains how to start using the component and describes its features.

Reviewer 2: ```suggestion
The <a href = "https://www.telerik.com/blazor-ui/popup" target="_blank">Blazor Popup component</a> helps you easily display a popup for a target element (anchor) in your application. You can use to Telerik UI for Blazor Popup to display additional information. This article explains how to start using the component and describes its features.
```

## Creating Blazor Popup

1. Add the `<TelerikPopup>` tag to a Razor file.

> 1. Obtain a `@ref` of the component.

Reviewer 2: Consider a more descriptive step.
```suggestion
1. Obtain the component reference through `@ref`.
```

1. Use the [Show](#popup-reference-and-methods) method to display the `<TelerikPopup>`.

>caption Basic configuration of the Telerik Popup for Blazor

````CSHTML
<TelerikPopup AnchorSelector=".popup-target" @ref="@PopupRef">
    I am a Telerik Popup.
</TelerikPopup>

<div class="popup-target styled-container" @onclick="@(_ => PopupRef.Show())">
    Popup target

> </div>

Reviewer 2: Better use our own components, for example a Button.

@code {
    private TelerikPopup PopupRef { get; set; }
}

<style>
    .styled-container {
        width: 200px;
        height: 30px;
        background-color: yellowgreen;
    }
</style>
````

## Popup Positioning and Collision

Use the available positioning and collision settings to customize how the Popup positions and reacts to insufficient space in the viewport. [Read more about the Blazor Popup Positioning and Collision...]({%slug popup-position-collision%})

## Popup Animations 

Use the available parameters to customize the animation type and its duration. [Read more about the Blazor Popup animations...]({%slug popup-animation%}).

## Popup Parameters

> The Blazor Popup provides parameters to configure the component. Also check the [Popup API Reference](/blazor-ui/api/Telerik.Blazor.Components.TelerikPopup) for a full list of properties.

Reviewer 2: ```suggestion
The Blazor Popup provides parameters to configure the component. Also check the [Popup API Reference](/blazor-ui/api/Telerik.Blazor.Components.TelerikPopup) for a full list of properties, methods and events.
```

| `VerticalAlign` | `PopupVerticalAlign ` enum <br /> (`Top`) | Determines if the Popup will touch the anchor with its top, bottom, or center. [Read more about Popup Positioning.]({%slug popup-position-collision%}) |
| `VerticalCollision` | `PopupCollision  ` enum <br /> (`Flip`) | Sets the behavior of the Popup when it doesn't fit in the viewport based on the vertical plane. [Read more about Popup collision behavior.]({%slug popup-position-collision%}) |
| `VerticalOffset` | `double` | Sets the vertical space between the Popup and its anchor in `pixels`. |

### Styling and Appearance

The following parameters enable you to customize the appearance of the Blazor Popup:

| Parameter | Type | Description |
| --- | --- | --- |
| `Class` | `string` | The custom CSS class to be rendered on the `<div>` element, which wraps the component `ChildContent`. Use for [styling customizations]({%slug themes-override%}). |
| `Height` | `string` | The height of the Popup. |
| `Width` | `string` | The width of the Popup. |

## Popup Reference and Methods

To execute Popup methods, obtain a reference to the component instance with `@ref`.

| Method  | Description |
|---------|-------------|

> | `Refresh` | Use this method to programmatically re-render the Popup. |

Reviewer 2: Provide more context and tips. See https://docs.telerik.com/blazor-ui/components/dialog/overview#dialog-reference-and-methods
